addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,163 @@
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                List<Channel> channels = getThing().getChannels();
+                for (Channel channel : channels) {
+                    if (isLinked( channel.getUID().getId() )) {
+                        publishSensorValue( channel.getUID() );
+                    }
+                }
+            }
+        };
+
+        scheduler.scheduleWithFixedDelay(refresher, 0, refreshTime.intValue(), TimeUnit.SECONDS);
+        logger.debug("Start automatic refresh every {} seconds", refreshTime.intValue());
+    }
+
+    private void publishSensorValue(ChannelUID channelUID) {
+        String channelID = channelUID.getId();
+        switch (channelID) {
+            case TEMPERATURE:
+                publishTemperatureSensorState(channelUID);
+                break;
+            default:
+                logger.debug("Can not update channel with ID : {} - channel name might be wrong!", channelID);
+                break;
+        }
+    }
+
+    private void publishTemperatureSensorState(ChannelUID channelUID) {
+        BigDecimal temp = readSensorTemperature(gpioBusFile);
+        if (temp != null) {
+            updateState(channelUID, new DecimalType(temp));
+        }
+    }
+
+    private BigDecimal readSensorTemperature(String gpioFile) {
+        try (Stream<String> stream = Files.lines(Paths.get(gpioFile))) {
+            Optional<String> temperatureLine = stream
+                    .filter(s -> s.contains(OneWireGPIOBindingConstants.FILE_TEMP_MARKER)).findFirst();
+            if (temperatureLine.isPresent()) {
+                String line = temperatureLine.get();
+                String tempString = line.substring(line.indexOf(OneWireGPIOBindingConstants.FILE_TEMP_MARKER)
+                        + OneWireGPIOBindingConstants.FILE_TEMP_MARKER.length());
+                Integer intTemp = Integer.parseInt(tempString);
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                return BigDecimal.valueOf(intTemp).movePointLeft(3);
+            } else {
+                logger.debug(
+                        "GPIO file didn't contain line with 't=' where temperature value should be available. Check if configuration points to the proper file");
+                return null;
+            }
+        } catch (IOException | InvalidPathException e) {
+            logger.debug("error reading GPIO bus file. File path is: {}.  Check if path is proper. {}", gpioFile, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error reading GPIO bus file.");
+            return null;
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        scheduler.shutdown();
```

Comment:
No, you must never shutdown the scheduler! This is used by all different bindings.
What you need to cancel is the scheduled job from line 109 (you get a handle as a return value there).

Reply:
ups. I wanted to do it to fast, and I didn't noticed that scheduler is global. Sorry for that, fixed :)